NBitcoin/Transaction.cs

```diff
@@ -1352,6 +1356,9 @@ public TxOutList Outputs
 		//Since it is impossible to serialize a transaction with 0 input without problems during deserialization with wit activated, we fit a flag in the version to workaround it
 		protected const uint NoDummyInput = (1 << 27);
 
+		public bool Equals(Transaction other) =>

```

Comment:
Q: If I have the same transaction twice, with and without witnesses, should this method return true or false? @NicolasDorier? 

I am thinking on this scenario: my application maintains its own mempool and that means that for each transaction it receives, it checks if it is already in the list of transactions using this `Equals(Transaction)` method. It could be possible(?) to receive the same transaction from two different nodes using different serialization format. In this case I think this should return true, otherwise i would end with duplicated transactions.

Reply:
We should remove this. Lot's of code depends on the fact that we are comparing reference here. We should keep it that way.